modules.d.ts

```diff
@@ -10,35 +10,247 @@ export declare const decodePresenceMessage: Types.PresenceMessageStatic['fromEnc
+ * ```
+ *
+ * When provided, you can control whether the client uses MessagePack via the {@link Types.ClientOptions.useBinaryProtocol} client option. If you do not provide this module, then the library will always JSON format for encoding messages.
+ */
 export declare const MsgPack: unknown;
+
+/**
+ * Provides a {@link BaseRealtime} instance with the ability to interact with a channel’s presence set.
+ *
+ * To create a client that includes this module, include it in the `ModulesMap` that you pass to the {@link BaseRealtime.constructor}:
+ *
+ * ```javascript
+ * import { BaseRealtime, WebSocketTransport, FetchRequest, RealtimePresence } from 'ably/modules';
+ * const realtime = new BaseRealtime(options, { WebSocketTransport, FetchRequest, RealtimePresence });
+ * ```
+ *
+ * If you do not provide this module, then attempting to access a channel’s {@link Types.RealtimeChannel.presence} property will cause a runtime error.
+ */
 export declare const RealtimePresence: unknown;
+
+/**
+ * Provides a {@link BaseRealtime} instance with the ability to establish a connection with the Ably realtime service using a [WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API) connection.
+ *
+ * To create a client that includes this module, include it in the `ModulesMap` that you pass to the {@link BaseRealtime.constructor}:
+ *
+ * ```javascript
+ * import { BaseRealtime, WebSocketTransport, FetchRequest } from 'ably/modules';
+ * const realtime = new BaseRealtime(options, { WebSocketTransport, FetchRequest });
+ * ```
+ *
+ * Note that network conditions, such as firewalls or proxies, might prevent the client from establishing a WebSocket connection. For this reason, you may wish to provide the `BaseRealtime` instance with the ability to alternatively establish a connection using a transport that is less susceptible to these external conditions. You do this by passing an alternative transport module, such as {@link XHRStreaming}, alongside `WebSocketTransport`:
+ *
+ * ```javascript
+ * import { BaseRealtime, WebSocketTransport, FetchRequest } from 'ably/modules';
+ * const realtime = new BaseRealtime(options, { WebSocketTransport, XHRStreaming, FetchRequest });
+ * ```
+ */
 export declare const WebSocketTransport: unknown;
+
+/**
+ * Provides a {@link BaseRealtime} instance with the ability to establish a connection with the Ably realtime service using the browser’s [XMLHttpRequest API](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest).
+ *
+ * ```javascript
+ * import { BaseRealtime, WebSocketTransport, FetchRequest } from 'ably/modules';
+ * const realtime = new BaseRealtime(options, { XHRPolling, FetchRequest });
+ * ```
+ *
+ * Provide this module if you wish the client to have an alternative mechanism for connecting to Ably if it’s unable to establish a WebSocket connection.
+ */
 export declare const XHRPolling: unknown;
+
+// TODO how is it different to XHRPolling? How should the user choose between the two?
```

Comment:
Do you have any knowledge / thoughts that would help here, @owenpearson?

Reply:
`xhr_streaming` uses http streaming, ie the request is held open for a long time (60s per request i believe) and realtime messages are sent as chunks, whereas `xhr_polling` is just long polling, ie the request is held open until the server responds and then a new request is opened immediately after.

HTTP streaming is sometimes blocked by corporate firewalls so `xhr_polling` is the more likely transport to succeed but is less preferable because it more requests.

Reply:
Thanks for the information. I've added some wording; see if you think it's enough?

Reply:
That looks great, thanks 👍